internal/services/springcloud/spring_cloud_gateway_sso_resource.go

```diff
@@ -0,0 +1,242 @@
+				return metadata.ResourceRequiresImport(s.ResourceType(), id)
+			}
+
+			existing.Properties.SsoProperties = &appplatform.SsoProperties{
+				Scope:        utils.ToPtr(model.Scope),
+				ClientID:     utils.String(model.ClientID),
+				ClientSecret: utils.String(model.ClientSecret),
+				IssuerURI:    utils.String(model.IssuerUri),
+			}
+
+			future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.SpringName, id.GatewayName, existing)
+			if err != nil {
+				return fmt.Errorf("creating/updating %s: %+v", id, err)
+			}
+
+			if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+				return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
+			}
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (s SpringCloudGatewaySsoResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model SpringCloudGatewaySsoModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.AppPlatform.GatewayClient
+			id, err := parse.SpringCloudGatewayID(model.SpringCloudGatewayId)
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.GatewayName)
+			if err != nil {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			existing.Properties.SsoProperties = &appplatform.SsoProperties{
+				Scope:        utils.ToPtr(model.Scope),
+				ClientID:     utils.String(model.ClientID),
+				ClientSecret: utils.String(model.ClientSecret),
+				IssuerURI:    utils.String(model.IssuerUri),
+			}
+
+			future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.SpringName, id.GatewayName, existing)
+			if err != nil {
+				return fmt.Errorf("creating/updating %s: %+v", id, err)
+			}
+
+			if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+				return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (s SpringCloudGatewaySsoResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model SpringCloudGatewaySsoModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+			client := metadata.Client.AppPlatform.GatewayClient
+
+			id, err := parse.SpringCloudGatewayID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.GatewayName)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+			state := SpringCloudGatewaySsoModel{
+				SpringCloudGatewayId: id.ID(),
```

Comment:
this field is available as `id` so there's no need to output this as a second field?

Reply:
This is not a computed field, it's a required field, so I must set its value to prevent diff.